### PR TITLE
fix the type information for routing policies in templates

### PR DESCRIPTION
**Pull request by ratulm**
1. Fix type information in templates for policies parameter
2. Make nodes and policies optional parameters in question constructor
3. Some cleanup

No functional change

### CODE REVIEW

File: projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java [hunk 1]
```diff
@@ -21,12 +21,10 @@
...
   private static final BgpRouteConstraints DEFAULT_ROUTE_CONSTRAINTS =
       BgpRouteConstraints.builder().build();
-  private static final String DEFAULT_NODES = "";
-  private static final String DEFAULT_POLICIES = "";
   private static final Action DEFAULT_ACTION = Action.PERMIT;
 
-  @Nonnull private final String _nodes;
-  @Nonnull private final String _policies;
+  @Nullable private final String _nodes;
+  @Nullable private final String _policies;
```
Comment (millstein) [inline projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java:27]: I don't understand the goal -- what is this allowing that wasn't allowed before?  It seems like these things were already optional on the Client side, but by allowing null internally you're just introducing more possibility for null pointer exceptions.

File: projects/question/src/main/java/org/batfish/question/testroutepolicies/TestRoutePoliciesQuestion.java [hunk 2]
```diff
@@ -24,23 +25,21 @@
...
-  private static final String DEFAULT_POLICY = "";
 
-  private final Direction _direction;
-  private final String _nodes;
-  private final String _policies;
-  private final List<BgpRoute> _inputRoutes;
+  @Nonnull private final Direction _direction;
+  @Nullable private final String _nodes;
+  @Nullable private final String _policies;
+  @Nonnull private final List<BgpRoute> _inputRoutes;
```
Comment (millstein) [inline projects/question/src/main/java/org/batfish/question/testroutepolicies/TestRoutePoliciesQuestion.java:32]: Same question here.

**Review by ratulm — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6241)* status: 0 of 9 files reviewed, 2 unresolved discussions (waiting on @anothermattbrown, @millstein, and @ratulm)

___
*[projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java, line 27 at r2](https://reviewable.io/reviews/batfish/batfish/6241#-MIDgwFELn8Bc78ZnoUQ-r2-27:-MIFyJIf4Ha99KWJba_W:b-pj2k7q) ([raw file](https://github.com/batfish/batfish/blob/574e3191f434fe060c3bfdcef5c686f9456a1d60/projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java#L27)):*
<details><summary><i>Previously, millstein (Todd Millstein) wrote…</i></summary><blockquote>

I don't understand the goal -- what is this allowing that wasn't allowed before?  It seems like these things were already optional on the Client side, but by allowing null internally you're just introducing more possibility for null pointer exceptions.
</blockquote></details>

The testroute policies wasn't allowing null values for nodes and policies (via checkNotNull in its json creator), which was what I went in to fix. 

The rest is bringing in line how we do most other questions -- retain the user input as is and let the answerer decide how it wants to treat null vs. empty string vs. non-empty string.

**Review by millstein — COMMENTED** [review]
> Reviewed 3 of 5 files at r1, 1 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6241)* status: 4 of 9 files reviewed, 2 unresolved discussions (waiting on @anothermattbrown and @ratulm)

___
*[projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java, line 27 at r2](https://reviewable.io/reviews/batfish/batfish/6241#-MIDgwFELn8Bc78ZnoUQ-r2-27:-MIGBT_TCYhBsMEjmLhc:b-j40y73) ([raw file](https://github.com/batfish/batfish/blob/574e3191f434fe060c3bfdcef5c686f9456a1d60/projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java#L27)):*
<details><summary><i>Previously, ratulm wrote…</i></summary><blockquote>

The testroute policies wasn't allowing null values for nodes and policies (via checkNotNull in its json creator), which was what I went in to fix. 

The rest is bringing in line how we do most other questions -- retain the user input as is and let the answerer decide how it wants to treat null vs. empty string vs. non-empty string.
</blockquote></details>

Got it.  Now that these allow null it'd be useful to add a test where the question has null values in those places.

**Review by ratulm — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6241)* status: 3 of 11 files reviewed, 2 unresolved discussions (waiting on @anothermattbrown, @millstein, and @ratulm)

___
*[projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java, line 27 at r2](https://reviewable.io/reviews/batfish/batfish/6241#-MIDgwFELn8Bc78ZnoUQ-r2-27:-MIGLaz6CBuNiYS6w6Fs:bnf3hms) ([raw file](https://github.com/batfish/batfish/blob/574e3191f434fe060c3bfdcef5c686f9456a1d60/projects/minesweeper/src/main/java/org/batfish/minesweeper/question/searchroutepolicies/SearchRoutePoliciesQuestion.java#L27)):*
<details><summary><i>Previously, millstein (Todd Millstein) wrote…</i></summary><blockquote>

Got it.  Now that these allow null it'd be useful to add a test where the question has null values in those places.
</blockquote></details>

ok, did a little refactoring of the answerers to reduce NPE chances and added tests.

**Review by ratulm — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6241)* status: 3 of 11 files reviewed, 2 unresolved discussions (waiting on @anothermattbrown and @millstein)

___
*[projects/question/src/main/java/org/batfish/question/testroutepolicies/TestRoutePoliciesQuestion.java, line 32 at r2](https://reviewable.io/reviews/batfish/batfish/6241#-MIDgwFELn8Bc78ZnoUS-r2-32:-MIGLm4r4WtxQzJB65_E:b-896fix) ([raw file](https://github.com/batfish/batfish/blob/574e3191f434fe060c3bfdcef5c686f9456a1d60/projects/question/src/main/java/org/batfish/question/testroutepolicies/TestRoutePoliciesQuestion.java#L32)):*
<details><summary><i>Previously, millstein (Todd Millstein) wrote…</i></summary><blockquote>

Same question here.
</blockquote></details>

Done.

**Review by millstein — COMMENTED** [review]
> Reviewed 3 of 5 files at r3.
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6241)* status: 6 of 11 files reviewed, all discussions resolved (waiting on @anothermattbrown)
